### PR TITLE
Add telemetry guide

### PR DESCRIPTION
**Pull request by hamishwillee**
This adds explanation of how to use the telemetry API. It uses just a few examples, but makes it clear that all the rest are the same, and points to the examples and integration test for more "real code".

The use cases I've tried to address are:
- Setting rate
- Reporting regular updates
- Reporting changes
- Blocking on state

@julianoes I am not particularly concerned that the code will be incorrect - more that the recommendations are not "best practise". Appreciate your review.

### CODE REVIEW

File: en/guide/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+All the methods of a particular type (synchronous, asynchronous, and set_rate methods) are used in the same way.
+
+## API Overview
+
+The `Telemetry` API provides methods to return the following types of information:
+
+* [Position](../api_reference/structdronecore_1_1_telemetry_1_1_position.md) - latitude and longitude in degrees, and altitude relative to sea level and to the takeoff altitude.
+* [Battery](../api_reference/structdronecore_1_1_telemetry_1_1_battery.md) - voltage and percentage power remaining.
+* [GroundSpeedNED](http://localhost:4000/en/api_reference/structdronecore_1_1_telemetry_1_1_ground_speed_n_e_d.md) - velocity components in NED co-ordinates.
```
Comment (julianoes) [inline en/guide/telemetry.md]: `co-ordinates` -> `coordinates`?

File: en/guide/telemetry.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+In addition there are a number of methods that return vehicle "state":
+* Current flight mode ([FlightModes](../api_reference/classdronecore_1_1_telemetry.md#classdronecore_1_1_telemetry_1a881d44b3a1522ea14bff8834edd4145a)).
+* Whether the vehicle is "all healthy" (aggregates the vehicle [Health](../api_reference/structdronecore_1_1_telemetry_1_1_health.md) information). This is used to check if the vehicle is *ready* to arm.
+* Whether the vehicle is armed/disarmed.
+* Whether the vehicle is flying/in air.
+
+
+## Setting the Update Rate {#update-rate}
+
+The telemetry update rate determines the frequency at which callbacks will be invoked with new information, and also the probable "freshness" of data obtained when using synchronous telemetry APIs. The default update rate depends on the autopilot and may also be limited by the characteristics of the communications channel. You can set the rate for *each* type of telemetry, and both synchronous or asynchronous rate-setting methods are provided.
```
Comment (julianoes) [inline en/guide/telemetry.md]: Thanks, that seems correct.

File: en/guide/telemetry.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+The asynchronous callbacks are updated every time new information is provided by the vehicle. For some types of telemetry you may only wish to report only when the value changes.
+
+The example below shows how to use [flight_mode_async()](../api_reference/classdronecore_1_1_telemetry.md#classdronecore_1_1_telemetry_1ac8842dec06db4bd54c8c2ba2deb0d34a) to report only when the current flight mode changes. The example uses a lambda function callback that captures a variable for the last mode. This variable is compared to the current flight mode to determine whether the value has changed and needs to be reported.
+
+```cpp
+// Set up callback to monitor flight mode 'changes' 
+Telemetry::FlightMode oldFlightMode=Telemetry::FlightMode::UNKNOWN;
+device.telemetry().flight_mode_async([&oldFlightMode](Telemetry::FlightMode flightMode) {
+    if (oldFlightMode!=flightMode) {
```
Comment (julianoes) [inline en/guide/telemetry.md]: Nitpick, I would format it this way:

```
if (oldFlightMode != flightMode) {
```

File: en/guide/telemetry.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+## Further Information
+
+Additional information/examples for the Telemetry API are linked below:
+
+* [DroneCore Examples](../examples/README.md)
+* Integration tests:
+  * [telemetry_async.cpp](https://github.com/dronecore/DroneCore/blob/master/integration_tests/telemetry_async.cpp)
+  * [telemetry_health.cpp](https://github.com/dronecore/DroneCore/blob/master/integration_tests/telemetry_health.cpp)
+  * [telemetry_modes.cpp](https://github.com/dronecore/DroneCore/blob/master/integration_tests/telemetry_modes.cpp)
+  * [telemetry_simple.cpp](https://github.com/dronecore/DroneCore/blob/master/integration_tests/telemetry_simple.cpp)
```
Comment (julianoes) [inline en/guide/telemetry.md]: Nice, good idea to link to the integration tests as well.

File: en/guide/telemetry.md (hunk 3 above)
Comment (hamishwillee) [reply en/guide/telemetry.md]: Thanks. I'm happy with nitpicking !

File: en/guide/telemetry.md (hunk 1 above)
Comment (hamishwillee) [reply en/guide/telemetry.md]: I'll change. FYI, both are correct, but the second is in wider usage.

**Pull request merged**